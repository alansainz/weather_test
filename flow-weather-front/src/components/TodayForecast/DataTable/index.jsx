import React from 'react';
import Row from './Row'
import { 
  Container,
  DataContainer,
  Header
} from './style.js'

const metersToKm = (meters) => meters / 1000;

const DataTable = ({ windSpeed, humidity, pressure, visibility }) => {
  const tableData = [
    {
      title: 'Wind',
      symbol: 'km/h',
      value: windSpeed && windSpeed
    },
    {
      title: 'Humidity',
      symbol: '%',
      value: humidity && humidity
    },
    {
      title: 'Pressure',
      symbol: 'mb',
      value: pressure && pressure
    },
    {
      title: 'Visibility',
      symbol: 'km',
      value: visibility && metersToKm(visibility),
      isLast: true
    }
  ]
  return (
    <Container>
      <DataContainer>
        <Header>Extra Info</Header>
        {
          tableData.map((row, index) => 
            <Row 
              key={index}
              title={row.title} 
              value={row.value} 
              symbol={row.symbol}
            />
          )
        }
      </DataContainer>
    </Container>
  )
};

export default DataTable