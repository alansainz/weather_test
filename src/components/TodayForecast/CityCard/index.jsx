import React from 'react';
import { returnTempOrDoubleDash } from '../../../utils';

import { 
  Container, 
  City, 
  Temperature, 
  DataContainer, 
  Description, 
  MaxMinTemperature 
} from './style.js'

const CityCard = ({ city, temperature, description, maxTemp, minTemp }) => (
  <Container>
    <City>
      { city && city.toUpperCase() }
    </City>
    
    <DataContainer>
      <Temperature>
        { returnTempOrDoubleDash(Math.floor(temperature)) }
      </Temperature>
      <Description>
        { description && description.toUpperCase() }
      </Description>
      <MaxMinTemperature>
        { `Max. ${returnTempOrDoubleDash(maxTemp)} Min. ${returnTempOrDoubleDash(minTemp)}` }
      </MaxMinTemperature>
    </DataContainer>
  </Container>
);

export default CityCard