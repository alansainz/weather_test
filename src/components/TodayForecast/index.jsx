import React from 'react';
import CityCard from './CityCard'
import DataTable from './DataTable'

import { Container } from './style.js'

const TodayForecast = (props) => (
  <Container>
    <CityCard {...props.cityCard} />
    <DataTable {...props.dataTable} />
  </Container>
)


export default TodayForecast