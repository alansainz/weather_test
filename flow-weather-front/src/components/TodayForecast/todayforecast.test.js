import React from 'react';
import ReactDOM from 'react-dom';
import TodayForecast from './index.jsx';
import { shallow  } from 'enzyme';

describe("TodayForecast Component", () => {
  it('should render with no props', () => {
    const component = shallow(<TodayForecast />);
    expect(component).toMatchSnapshot();
  });

  it('should render with given props', () => { 
    const todayData = {
      cityCard: {
        city: 'test name',
        description: 'test description',
        temperature: 12,
        maxTemp: 20,
        minTemp: 1,
      },
      dataTable: {
        windSpeed: 1000,
        humidity: 50,
        pressure: 100,
        visibility: 23
      }
    }
    const component = shallow(<TodayForecast {...todayData} />);
    expect(component).toMatchSnapshot();
  });
})