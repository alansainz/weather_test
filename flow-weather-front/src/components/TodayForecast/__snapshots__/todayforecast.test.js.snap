// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodayForecast Component should render with given props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayForecast
    cityCard={
      Object {
        "city": "test name",
        "description": "test description",
        "maxTemp": 20,
        "minTemp": 1,
        "temperature": 12,
      }
    }
    dataTable={
      Object {
        "humidity": 50,
        "pressure": 100,
        "visibility": 23,
        "windSpeed": 1000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CityCard
          city="test name"
          description="test description"
          maxTemp={20}
          minTemp={1}
          temperature={12}
        />,
        <DataTable
          humidity={50}
          pressure={100}
          visibility={23}
          windSpeed={1000}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "city": "test name",
          "description": "test description",
          "maxTemp": 20,
          "minTemp": 1,
          "temperature": 12,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "humidity": 50,
          "pressure": 100,
          "visibility": 23,
          "windSpeed": 1000,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gZMcBi",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: row
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gZMcBi",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CityCard
            city="test name"
            description="test description"
            maxTemp={20}
            minTemp={1}
            temperature={12}
          />,
          <DataTable
            humidity={50}
            pressure={100}
            visibility={23}
            windSpeed={1000}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "city": "test name",
            "description": "test description",
            "maxTemp": 20,
            "minTemp": 1,
            "temperature": 12,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "humidity": 50,
            "pressure": 100,
            "visibility": 23,
            "windSpeed": 1000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: row
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TodayForecast Component should render with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayForecast />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CityCard />,
        <DataTable />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gZMcBi",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: row
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gZMcBi",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CityCard />,
          <DataTable />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: row
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
